src/Compilers/Core/Portable/Operations/Expressions.cs

```diff
@@ -1,405 +1,116 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System.Collections.Immutable;
-using System.Diagnostics;
 using System.Linq;
 
 namespace Microsoft.CodeAnalysis.Semantics
 {
-    internal sealed class ConditionalChoice : IConditionalChoiceExpression
+    internal sealed partial class ConditionalChoiceExpression : IConditionalChoiceExpression
     {
-        public ConditionalChoice(IOperation condition, IOperation ifTrue, IOperation ifFalse, ITypeSymbol resultType, SyntaxNode syntax)
+        public ConditionalChoiceExpression(IOperation condition, IOperation ifTrue, IOperation ifFalse, ITypeSymbol resultType, SyntaxNode syntax) :
+            this(condition,
+                ifTrue,
+                ifFalse,
+                IsInvalidOperation(condition, ifTrue, ifFalse, resultType),
+                syntax,
+                resultType,
+                default(Optional<object>))
         {
-            this.Condition = condition;
-            this.IfTrueValue = ifTrue;
-            this.IfFalseValue = ifFalse;
-            this.Type = resultType;
-            this.Syntax = syntax;
         }
 
-        public IOperation Condition { get; }
-
-        public IOperation IfTrueValue { get; }
-
-        public IOperation IfFalseValue { get; }
-
-        public ITypeSymbol Type { get; }
-
-        public SyntaxNode Syntax { get; }
-
-        public OperationKind Kind => OperationKind.ConditionalChoiceExpression;
-
-        public bool IsInvalid => Condition == null || Condition.IsInvalid || IfTrueValue == null || IfTrueValue.IsInvalid || IfFalseValue == null || IfFalseValue.IsInvalid || Type == null;
-
-        public Optional<object> ConstantValue => default(Optional<object>);
-
-        public void Accept(OperationVisitor visitor)
-        {
-            visitor.VisitConditionalChoiceExpression(this);
-        }
-
-        public TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-        {
-            return visitor.VisitConditionalChoiceExpression(this, argument);
-        }
-    }
-
-    internal sealed class Assignment : IExpressionStatement
-    {
-        private readonly AssignmentExpression _assignment;
-
-        public Assignment(IOperation target, IOperation value, SyntaxNode syntax)
+        private static bool IsInvalidOperation(IOperation condition, IOperation ifTrue, IOperation ifFalse, ITypeSymbol resultType)
```

Comment:
No.  I see how it's being used now.  It's just unfortunate.  I think we'd like to get rid of the whole 'IsInvalid' portion of the API (*eventually*).